src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -34,7 +34,11 @@ public WinInetProxyHelper()
                         AutoDetect,
                         Proxy,
                         ProxyBypass);
-                    _useProxy = true;
+
+                    // WinHttpGetIEProxyConfigForCurrentUser has returned settings where "Use automatic configuration script" and
+                    // "Automatically detect settings" are unchecked. It means that WPAD protocol shouldn't be used, and there is
+                    // no explicit PAC file set. We can avoid calling WinHttpGetProxyForUrl for each request.
+                    _useProxy = (AutoConfigUrl == null && AutoDetect == false) ? false : true;
```

Comment:
I don't see how this PR is a replacement/improvement on PR #28859.

This change will break WinHttpHandler.  Basically this line above is turning off using a proxy for WinHttpHandler  But that isn't correct.  Just because the "Automatic detect settings" or "Use automatic configuration script" is turned off, it doesn't mean not to use a proxy.

This change in WinInetProxyHelper was originally motivated by needing to improve the performance of SocketsHttpHandler.  That handler has separate code to manually parse the proxy bypass list in the IE settings as well as use an explicitly specified proxy from the IE settings.  But WinHttpHandler doesn't have that.  When this class, WinInetProxyHelper, is used (on Windows 7 for example), it is the only thing that can parse the results from reading the IE settings information.

Reply:
@caesar1995 and I discussed this offline. It turns out that WinHttpHandler is passing the manual settings directly into the WinHttpOpen() call before making a request.  So, this change by itself doesn't completely break WinHttpHandler.

However, we discussed the fact that this change still mutates states in WinInetProxyHelper and that affects how WinHttpHandler and WinHTTP will work when both "Automatic detect settings" are "Use automatic configuration script" checked.

The suggestion I made is for this new optimization to be separated and only affect SocketsHttpHandler and not affect WinHttpHandler logic.